From cdef058e8e14b16028d29f40d730f3970f9a2c70 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 22 Sep 2021 18:43:05 +0300
Subject: [PATCH] [SW-0] gaudi2: arc: use run arc function

This patch uses run arc function to set arc0 in running state
as part of preboot_start_image function.

Change-Id: I481faea3aa1d0809fc39321d2c1ab6ef3ec3e836
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index 087a8ca..61f3107 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -152,17 +152,8 @@
 		regmap_writel(0x10111, mmCPU_CA53_CFG_ARM_RST_CONTROL);
 	}
 
-	/*
-	 * Set ARC0 in running state
-	 * 1. Set run bit in RUN_HALT_REQ register
-	 * 2. Wait for ACK from core in RUN_HALT_ACK register
-	 * 3. Clear run bit in RUN_HALT_REQ register
-	 */
-	regmap_writel(BIT(0), mmPSOC_ARC0_CFG_RUN_HALT_REQ);
-	while (!(regmap_readl(mmPSOC_ARC0_CFG_RUN_HALT_ACK) &
-				PSOC_ARC0_CFG_RUN_HALT_ACK_RUN_ACK_MASK))
-		;
-	regmap_writel(0x0, mmPSOC_ARC0_CFG_RUN_HALT_REQ);
+	/* Set ARC0 in running state */
+	run_arc(0);
 
 	/*
 	 * Jump to __reset of zephyr. After image_entry, control is not
